flask_mongorest/resources.py

```diff
@@ -199,15 +200,24 @@ def serialize_field(self, obj, **kwargs):
         else:
             return self.serialize(obj, **kwargs)
 
+    def _subresource_class(self, obj):
+        default_s_class = self._child_document_resources[self.default_child_resource_document] if self.default_child_resource_document else None
+        s_class = self._child_document_resources.get(obj.__class__, default_s_class)
+        return s_class if s_class != self.__class__ else None
+
+    def _create_subresource(self, obj):
+        r = self._subresource_class(obj)()
+        r.data = self.data
+        return r
+
     def serialize(self, obj, **kwargs):
         if not obj:
             return {}
 
         # If a subclass of an obj has been called with a base class' resource,
         # use the subclass-specific serialization
-        if obj.__class__ in self._child_document_resources \
-        and self._child_document_resources[obj.__class__] != self.__class__:
-            return obj and self._child_document_resources[obj.__class__]().serialize_field(obj, **kwargs)
+        if self._subresource_class(obj):
+            return self._create_subresource(obj).serialize(obj, **kwargs)
```

Comment:
We end up calling `self._subresource_class(obj)` twice here (once directly, once through `_create_subresource`). Ideally we would only call it once. Same in the other methods.